### PR TITLE
a new version of google analytics plugin using latest code

### PR DESCRIPTION
**Pull request by sycom**
Hello,

i'm not very good with git, so my commits are a bit messy, but i managed to give something quiet polished. The new version of the plugin submitted seems to work fine. I'm not sure about the way of doing pull requests since it's one of my firsts. So don't be too cruel with me...

The proposed new version of the googleanalytics plugin is built with latest version (_i s o g r a m_) of google tracking code. I also added a settings tab for easier ga_account and ga_domain settings.

greatings

### CODE REVIEW

File: plugins/tiddlywiki/googleanalytics/readme.tid [hunk 1]
```diff
@@ -2,4 +2,5 @@ title: $:/plugins/tiddlywiki/googleanalytics/readme
 
 This plugin enables you to use Google Analytics to track access to your online TiddlyWiki document.
 
-[[Source code|https://github.com/Jermolene/TiddlyWiki5/blob/master/plugins/tiddlywiki/googleanalytics]]
+* [[Current plugin source|https://framagit.org/sycom/TiddlyWiki5/tree/plugin-googleanalytics]]
```
Comment (tobibeer) [inline plugins/tiddlywiki/googleanalytics/readme.tid]: Are you suggesting to change the source of the repo?

File: plugins/tiddlywiki/googleanalytics/plugin.info [hunk 2]
```diff
@@ -2,6 +2,8 @@
 	"title": "$:/plugins/tiddlywiki/googleanalytics",
 	"description": "Google Analytics: website statistics",
 	"author": "JeremyRuston",
+	"contributor": "Sylvain Comte",
+	"version":"5.1.14",
 	"core-version": ">=5.0.0",
-	"list": "readme usage"
+	"list": "settings readme usage"
```
Comment (tobibeer) [inline plugins/tiddlywiki/googleanalytics/plugin.info]: readme should always be first

File: plugins/tiddlywiki/googleanalytics/googleanalytics.js [hunk 3]
```diff
@@ -17,19 +17,23 @@ exports.name = "google-analytics";
...
+		GA_DOMAIN = $tw.wiki.getTiddlerText("$:/GoogleAnalyticsDomain").replace(/\n/g,"");
+	if (GA_DOMAIN == "" || GA_DOMAIN == undefined) GA_DOMAIN = "auto";
+
+	// using ga "isogram" function
+  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
+  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
+  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
+  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
+
+console.log("tiddlywiki analytics working with account :"+GA_ACCOUNT+", and domain :"+GA_DOMAIN+".");
```
Comment (tobibeer) [inline plugins/tiddlywiki/googleanalytics/googleanalytics.js]: I don't think a console.log should stay in here, should it?

File: plugins/tiddlywiki/googleanalytics/googleanalytics.js [hunk 4]
```diff
@@ -17,19 +17,23 @@ exports.name = "google-analytics";
...
-	ga.async = true;
-	ga.src = ("https:" == document.location.protocol ? "https://ssl" : "http://www") + ".google-analytics.com/ga.js";
-	document.body.appendChild(ga);
+	// getting parameters
+	GA_ACCOUNT = $tw.wiki.getTiddlerText("$:/GoogleAnalyticsAccount").replace(/\n/g,""),
+		GA_DOMAIN = $tw.wiki.getTiddlerText("$:/GoogleAnalyticsDomain").replace(/\n/g,"");
+	if (GA_DOMAIN == "" || GA_DOMAIN == undefined) GA_DOMAIN = "auto";
+
+	// using ga "isogram" function
+  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
```
Comment (tobibeer) [inline plugins/tiddlywiki/googleanalytics/googleanalytics.js]: minified code? where's the non-minified version?

File: plugins/tiddlywiki/googleanalytics/googleanalytics.js [hunk 5]
```diff
@@ -17,19 +17,23 @@ exports.name = "google-analytics";
 exports.platforms = ["browser"];
 exports.synchronous = true;
 
-var GOOGLE_ANALYTICS_ACCOUNT = "$:/GoogleAnalyticsAccount",
-	GOOGLE_ANALYTICS_DOMAIN = "$:/GoogleAnalyticsDomain";
+var GA_ACCOUNT,GA_DOMAIN;
```
Comment (tobibeer) [inline plugins/tiddlywiki/googleanalytics/googleanalytics.js]: why declare those variables here?

File: plugins/tiddlywiki/googleanalytics/googleanalytics.js [hunk 6]
```diff
@@ -1,9 +1,9 @@
-/*\
+﻿/*\
 title: $:/plugins/tiddlywiki/googleanalytics/googleanalytics.js
 type: application/javascript
 module-type: startup
 
-Runs Google Analytics with the account number in the tiddler `$:/GoogleAnalyticsAccount` and the domain name in `$:/GoogleAnalyticsDomain`
+Runs new "i s o g r a m" Google Analytics with the account number in the tiddler `$:/GoogleAnalyticsAccount` and the domain name in `$:/GoogleAnalyticsDomain`
```
Comment (tobibeer) [inline plugins/tiddlywiki/googleanalytics/googleanalytics.js]: `i s o g r a m`?

why the "rebranding"?

File: editions/tw5.com/tiddlers/system/GoogleAnalyticsAccount.tid [hunk 7]
```diff
@@ -1,3 +1,3 @@
 title: $:/GoogleAnalyticsAccount
 
-UA-32839735-1
\ No newline at end of file
+UA-32839735-1
```
Comment (tobibeer) [inline editions/tw5.com/tiddlers/system/GoogleAnalyticsAccount.tid]: why changed?

File: plugins/tiddlywiki/googleanalytics/readme.tid (hunk 1 above)
Comment (sycom) [inline plugins/tiddlywiki/googleanalytics/readme.tid]: Definitely not. First of my mistakes :-/ I did just as my merge is already rejected... This one can be blamed...

File: plugins/tiddlywiki/googleanalytics/googleanalytics.js (hunk 6 above)
Comment (sycom) [inline plugins/tiddlywiki/googleanalytics/googleanalytics.js]: For now there are two versions of tracker. But I did not see any version reference in Google documentation. Just the strong suggest to switch to the new one.

File: plugins/tiddlywiki/googleanalytics/plugin.info (hunk 2 above)
Comment (sycom) [inline plugins/tiddlywiki/googleanalytics/plugin.info]: Oh. OK. Did not read it in the docs I've red.

File: plugins/tiddlywiki/googleanalytics/googleanalytics.js (hunk 4 above)
Comment (sycom) [inline plugins/tiddlywiki/googleanalytics/googleanalytics.js]: Well, I don't know. I'll have to check if Google discloses it.

File: plugins/tiddlywiki/googleanalytics/googleanalytics.js (hunk 5 above)
Comment (sycom) [inline plugins/tiddlywiki/googleanalytics/googleanalytics.js]: Well, I don't really know. Is it bad?

File: editions/tw5.com/tiddlers/system/GoogleAnalyticsAccount.tid (hunk 7 above)
Comment (sycom) [inline editions/tw5.com/tiddlers/system/GoogleAnalyticsAccount.tid]: My text editor is badly adding newlines at end of files. No real wish to do this. I believed to have change this when rebasing. I'm very bad at git...

Comment (sycom) [inline editions/tw5.com/tiddlers/system/GoogleAnalyticsAccount.tid]: And many thanks for your patience.

**Pull request closed without merging**